libs/server/Storage/Functions/MainStore/VarLenInputMethods.cs

```diff
@@ -218,45 +214,61 @@ public int GetRMWModifiedValueLength(ref SpanByte t, ref SpanByte input)
-                        return sizeof(int) + t.Length + input.MetadataSize;
+                    case RespCommand.EXPIREAT:
+                    case RespCommand.PEXPIREAT:
+                        return sizeof(int) + t.Length + sizeof(long);
 
                     case RespCommand.SETRANGE:
-                        var offset = *((int*)(inputPtr + RespInputHeader.Size));
-                        var newValueSize = *((int*)(inputPtr + RespInputHeader.Size + sizeof(int)));
+                        var offset = input.parseState.GetInt(input.parseStateFirstArgIdx);
+                        var newValue = input.parseState.GetArgSliceByRef(input.parseStateFirstArgIdx + 1).ReadOnlySpan;
 
-                        if (newValueSize + offset > t.LengthWithoutMetadata)
-                            return sizeof(int) + newValueSize + offset + t.MetadataSize;
+                        if (newValue.Length + offset > t.LengthWithoutMetadata)
+                            return sizeof(int) + newValue.Length + offset + t.MetadataSize;
                         return sizeof(int) + t.Length;
 
                     case RespCommand.GETDEL:
                         // No additional allocation needed.
                         break;
 
                     case RespCommand.APPEND:
-                        var valueLength = *((int*)(inputPtr + RespInputHeader.Size));
+                        var valueLength = input.parseState.GetArgSliceByRef(input.parseStateFirstArgIdx).Length;
                         return sizeof(int) + t.Length + valueLength;
 
                     default:
-                        if (cmd >= 200)
+                        if ((byte)cmd >= CustomCommandManager.StartOffset)
                         {
-                            var functions = functionsState.customCommands[cmd - 200].functions;
+                            var functions = functionsState.customCommands[(byte)cmd - CustomCommandManager.StartOffset].functions;
                             // compute metadata for result
-                            int metadataSize = input.ExtraMetadata switch
+                            var metadataSize = input.arg1 switch
                             {
                                 -1 => 0,
                                 0 => t.MetadataSize,
                                 _ => 8,
                             };
-                            return sizeof(int) + metadataSize + functions.GetLength(t.AsReadOnlySpan(), input.AsReadOnlySpan().Slice(RespInputHeader.Size));
+                            return sizeof(int) + metadataSize + functions.GetLength(t.AsReadOnlySpan(), ref input);
                         }
                         throw new GarnetException("Unsupported operation on input");
                 }
             }
 
-            return sizeof(int) + input.Length - RespInputHeader.Size;
+            return sizeof(int) + input.parseState.GetArgSliceByRef(input.parseStateFirstArgIdx).ReadOnlySpan.Length +
```

Comment:
Why do we need to get the ReadOnlySpan first before finding the length? can try to avoid this.